### PR TITLE
docs: Example to install feast on local computer using Kind

### LINKED ISSUE #4527: Add example of deployment of Feast in local Kubernetes environment
As a data scientist I want to be able to install feast on a Kubernetes platform in my local computer and validate that it works.

### PR DESCRIPTION
**Pull request by dmartinol**
# What this PR does / why we need it:
Provide a reference implementation of Feast project installed on local Kind cluster and validated with a remote client.

* Prerequisites are mentioned and reference installation commands provided for MacOS.
* A `postgres` template is used to initialize the project.
* All the commands are executed from individual notebooks, which also report the output of the latest run.
* Hints are provided to customize the provided code for a personalized experience.
* A dedicated `Job` is used to apply the feature store definitions (e.g. `feast apply`) and materializations, overcoming issues: https://github.com/feast-dev/feast/issues/4526 and https://github.com/feast-dev/feast/issues/4529

**Note**: the execution of the client application is still affected by https://github.com/feast-dev/feast/issues/4392 until we release the new version

# Which issue(s) this PR fixes:
Fixes #4527

### CODE REVIEW

**Comment by dmartinol** [comment]
@tokoko @franciscojavierarceo

**Comment by lokeshrangineni** [comment]
A brief `readme.md` description about the example at https://github.com/dmartinol/feast/tree/install_kind/examples/kind-quickstart will help before looking at all the jupyter notebooks and figure it out. something similar to this PR description.

File: examples/kind-quickstart/01-Install.ipynb [hunk 1]
```diff
@@ -0,0 +1,932 @@
...
+      "Requirement already satisfied: locket in /Users/dmartino/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from partd>=1.2.0->dask>=2024.2.1->dask[dataframe]>=2024.2.1->feast==0.40.1) (1.0.0)\n",
+      "Requirement already satisfied: six>=1.5 in /Users/dmartino/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from python-dateutil>=2.8.2->pandas<3,>=1.4.3->feast==0.40.1) (1.16.0)\n",
+      "Requirement already satisfied: anyio<5,>=3.4.0 in /Users/dmartino/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from starlette<0.39.0,>=0.37.2->fastapi>=0.68.0->feast==0.40.1) (4.4.0)\n",
+      "Requirement already satisfied: sniffio>=1.1 in /Users/dmartino/.pyenv/versions/3.11.9/lib/python3.11/site-packages (from anyio<5,>=3.4.0->starlette<0.39.0,>=0.37.2->fastapi>=0.68.0->feast==0.40.1) (1.3.1)\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# WE MUST ENSURE PUTHON CONSISTENCY BETWEEN NOTEBOOK AND FEAST SERVERS\n",
```
Comment (lokeshrangineni) [inline examples/kind-quickstart/01-Install.ipynb:80]: may be typo - `PYTHON`

**Comment by dmartinol** [comment]
@franciscojavierarceo could pls you TAL?

File: examples/kind-quickstart/02-Client.ipynb [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+   ],
+   "source": [
+    "!cd client && python test_workflow.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note If you see the following error, it is likely due to the [issue #4392](https://github.com/feast-dev/feast/issues/4392):\n",
```
Comment (franciscojavierarceo) [inline examples/kind-quickstart/02-Client.ipynb:541]: Looks like you're showing this error on purpose then, eh?

Comment (dmartinol) [reply examples/kind-quickstart/02-Client.ipynb:541]: Yep, the issue has been fixed after this PR, I should have updated it.

**Pull request merged**